src/API/Management/Emails.php

```diff
@@ -8,65 +8,90 @@
-
-      return ResponseMediator::getContent($response);
-  }
-
-  /**
-   * @param array $data
-   *
-   * @return mixed
-   */
-  public function updateEmailProvider($data)
-  {
-      $response = $this->httpClient->patch('/emails/provider', [], json_encode($data));
-
-      return ResponseMediator::getContent($response);
-  }
-
-  /**
-   * @return mixed
-   */
-  public function deleteEmailProvider()
-  {
-      $response = $this->httpClient->delete('/emails/provider');
-
-      return ResponseMediator::getContent($response);
-  }
+     * @link https://auth0.com/docs/api/management/v2#!/Emails/get_provider
+     *
+     * @param null|string|array $fields
+     * @param null|string|array $includeFields
+     *
+     * @return mixed
+     */
+    public function getEmailProvider($fields = null, $includeFields = null)
+    {
+        $queryParams = [];
+        if ($fields !== null) {
+            if (is_array($fields)) {
+                $fields = implode(',', $fields);
+            }
+            $queryParams['fields'] = $fields;
+        }
+
+        if ($includeFields !== null) {
+            $queryParams['include_fields'] = $includeFields;
+        }
+
+        $query = '';
+        if (!empty($queryParams)) {
+            $query = '?'.http_build_query($queryParams);
+        }
+        $response = $this->httpClient->get('/emails/provider'.$query);
+
+        if (200 === $response->getStatusCode()) {
+            return ResponseMediator::getContent($response);
+        }
+
+        $this->handleExceptions($response);
+    }
+
+    /**
+     * @link https://auth0.com/docs/api/management/v2#!/Emails/post_provider
+     *
+     * @param array $data
+     *
+     * @return mixed
+     */
+    public function configureEmailProvider($data)
+    {
+        $response = $this->httpClient->post('/emails/provider', [], json_encode($data));
+
+        if (200 === $response->getStatusCode()) {
+            return ResponseMediator::getContent($response);
+        }
+
+        $this->handleExceptions($response);
+    }
+
+    /**
+     * @link https://auth0.com/docs/api/management/v2#!/Emails/patch_provider
+     *
+     * @param array $data
+     *
+     * @return mixed
+     */
+    public function updateEmailProvider($data)
+    {
+        $response = $this->httpClient->patch('/emails/provider', [], json_encode($data));
+
+        if (200 === $response->getStatusCode()) {
+            return ResponseMediator::getContent($response);
+        }
+
+        $this->handleExceptions($response);
+    }
+
+    /**
+     * @https://auth0.com/docs/api/management/v2#!/Emails/delete_provider
+     *
+     * @return mixed
+     */
+    public function deleteEmailProvider()
+    {
+        $response = $this->httpClient->delete('/emails/provider');
+
+        // TODO verify with Auth0 team that this is correct
+        if (204 === $response->getStatusCode()) {
+            return ResponseMediator::getContent($response);
```

Comment:
I've raised this one to be documented, it might get removed, if it does we can remove from API in another PR.